src/ChildProcess/Adapter.php

```diff
@@ -130,8 +130,8 @@ public function callFilesystem($function, $args, $errorResultCode = -1)
     {
         return $this->pool->rpc(Factory::rpc($function, $args))->then(function (Payload $payload) {
             return \React\Promise\resolve($payload->getPayload());
-        }, function (Payload $payload) {
-            return \React\Promise\reject(new Exception($payload->getPayload()['error']['error']['message']));
+        }, function ($payload) {
+            return \React\Promise\reject(new Exception($payload['error']['message']));
```

Comment:
The title suggests there must be a way to verify this is "the correct way"? :shipit:

Reply:
Yeah using array access and ditching the extra `error` index solve the errors.

(Also updated the title.)

Reply:
Is this tested or can we add some tests for this? :+1:

Reply:
Added a test for this :+1: